src/plugins/vis_default_editor/public/components/controls/palette_picker.tsx

```diff
@@ -36,24 +51,12 @@ export function PalettePicker<ParamName extends string>({
         fullWidth
         data-test-subj="visEditorPalettePicker"
         compressed
-        palettes={palettes
-          .getAll()
-          .filter(({ internal }) => !internal)
-          .map(({ id, title, getCategoricalColors }) => {
-            return {
-              value: id,
-              title,
-              type: 'fixed',
-              palette: getCategoricalColors(
-                10,
-                id === activePalette?.name ? activePalette?.params : undefined
-              ),
-            };
-          })}
+        palettes={palettesList}
         onChange={(newPalette) => {
+          const palette = palettesList.find((item) => item.value === newPalette);
           setPalette(paramName, {
             type: 'palette',
-            name: newPalette,
+            name: palette?.value ?? 'clear',
```

Comment:
What does the fallback to clear does here?

Reply:
Updated the code.

This line is updating the color of the chart for default palette: https://github.com/elastic/kibana/blob/18514cebaff7d2d94e313258d9fd1b8e5c784c5c/src/plugins/vis_default_editor/public/components/controls/palette_picker.tsx#L61

This line is updating the palette in palette picker to default if such is not found: https://github.com/elastic/kibana/blob/18514cebaff7d2d94e313258d9fd1b8e5c784c5c/src/plugins/vis_default_editor/public/components/controls/palette_picker.tsx#L64

Reply:
Perfect! Thank you :)